packages/components/src/range-control/index.js

```diff
@@ -26,19 +27,52 @@ function RangeControl( {
 	help,
 	allowReset,
 	initialPosition,
+	min,
+	max,
+	setState,
 	...props
 } ) {
 	const id = `inspector-range-control-${ instanceId }`;
-	const resetValue = () => onChange();
+	const currentInputValue = currentInput === null ? value : currentInput;
+	const resetValue = () => {
+		resetCurrentInput();
+		onChange();
+	};
+	const resetCurrentInput = () => {
+		// If the input is an empty string call onChange with an undefined argument to reset to value.
+		if ( currentInput === '' ) {
+			onChange();
+		}
+		if ( currentInput !== null ) {
+			setState( {
+				currentInput: null,
+			} );
+		}
+	};
+
 	const onChangeValue = ( event ) => {
 		const newValue = event.target.value;
-		if ( newValue === '' ) {
-			resetValue();
+		const newNumericValue = Number( newValue );
+		// If the input value is invalid temporarly save it to the state,
+		// without calling on change.
+		if (
+			( ! newNumericValue && newValue !== '0' ) ||
```

Comment:
It took me a bit to figure out why we were doing it this way. I guess `Number( '' )` coerces to `0`, so we're differentiating?

Alternatively, we might consider a stricter parse, checking that the produced value is not `NaN` ([`isNaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN)).

For example, `parseInt( '', 10 )` will produce `NaN`.

The main benefit I think is readability. Thoughts?

Reply:
Hi @aduth, 
> It took me a bit to figure out why we were doing it this way. I guess Number( '' ) coerces to 0, so we're differentiating?

Exactly, the idea was to differentiate. I agree with your suggestion, using parseInt( '', 10 ) and isNaN improves readability and I updated the code.